code/modules/admin/sql_message_system.dm

```diff
@@ -443,9 +495,15 @@
 			output += "<a href='?_src_=holder;[HrefToken()];showmessageckey=[index_ckey]'>[index_key]</a><br>"
 		qdel(query_list_messages)
 	else if(!type && !target_ckey && !index)
-		output += "<center></a> <a href='?_src_=holder;[HrefToken()];addmessageempty=1'>\[Add message\]</a><a href='?_src_=holder;[HrefToken()];addwatchempty=1'>\[Add watchlist entry\]</a><a href='?_src_=holder;[HrefToken()];addnoteempty=1'>\[Add note\]</a></center>"
+		output += "<center><a href='?_src_=holder;[HrefToken()];addmessageempty=1'>Add message</a><a href='?_src_=holder;[HrefToken()];addwatchempty=1'>Add watchlist entry</a><a href='?_src_=holder;[HrefToken()];addnoteempty=1'>Add note</a></center>"
 		output += ruler
-	usr << browse({"<!DOCTYPE html><html><head><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" /></head><body>[jointext(output, "")]</body></html>"}, "window=browse_messages;size=900x500")
+	var/datum/browser/browser = new(usr, "Note panel", "Manage player notes", 1000, 500)
+	usr << browse_rsc('html/high_button.png')
+	usr << browse_rsc('html/medium_button.png')
+	usr << browse_rsc('html/minor_button.png')
+	usr << browse_rsc('html/none_button.png')
```

Comment:
You should add these to an asset cache instead of running `browse_rsc` every time this window is opened.